subgraphs/gains-trade/src/mappings/helpers.ts

```diff
@@ -25,3 +48,279 @@ export function createTokenAmountArray(
+  fundingRatePerDay: BigDecimal,
+  event: ethereum.Event
+): void {
+  const leveragedAmount = collateralAmount.times(leverage);
+  pool.addInflowVolumeByToken(collateralToken, leveragedAmount);
+  pool.addVolumeByToken(collateralToken, leveragedAmount);
+
+  const pairStorageContract = PairStorage.bind(
+    NetworkConfigs.getPairStorageAddress()
+  );
+  const openFeePCall = pairStorageContract.try_pairOpenFeeP(pairIndex);
+  if (openFeePCall.reverted) {
+    log.error("[openFeePCall reverted] hash: {} pairIndex: {}", [
+      event.transaction.hash.toHexString(),
+      pairIndex.toString(),
+    ]);
+    return;
+  }
+  const openFeeP = openFeePCall.value;
+  const openingFee = bigDecimalToBigInt(
+    leveragedAmount
+      .times(openFeeP)
+      .toBigDecimal()
+      .div(PRECISION.times(BIGINT_HUNDRED).toBigDecimal())
+  );
+  pool.addPremiumByToken(
+    collateralToken,
+    openingFee,
+    TransactionType.COLLATERAL_IN
+  );
+
+  const positionID = position.getBytesID();
+  const collateralAmounts = createTokenAmountArray(
+    pool,
+    [collateralToken],
+    [collateralAmount]
+  );
+  account.collateralIn(pool, positionID, collateralAmounts, sharesTransferred);
+
+  if (leverage > BIGINT_ONE) {
+    account.borrow(
+      pool,
+      positionID,
+      Address.fromBytes(collateralToken.id),
+      leveragedAmount.minus(collateralAmount)
+    );
+  }
+
+  position.setLeverage(leverage.toBigDecimal());
+  position.setBalance(collateralToken, collateralAmount);
+  position.setCollateralBalance(collateralToken, collateralAmount);
+  position.addCollateralInCount();
+  position.setFundingrateOpen(fundingRatePerDay);
+}
+
+export function closeTrade(
+  pool: Pool,
+  account: Account,
+  position: Position,
+  pairIndex: BigInt,
+  collateralToken: Token,
+  collateralAmount: BigInt,
+  leverage: BigInt,
+  sharesTransferred: BigInt,
+  fundingRatePerDay: BigDecimal,
+  percentProfit: BigInt,
+  isExistingOpenPosition: boolean,
+  event: ethereum.Event,
+  isLiquidation: boolean,
+  liquidator: Address | null = null,
+  liquidatee: Address | null = null
+): void {
+  const leveragedAmount = collateralAmount.times(leverage);
+  const pnl = collateralAmount
+    .times(percentProfit)
+    .toBigDecimal()
+    .div(PRECISION.times(BIGINT_HUNDRED).toBigDecimal());
+  const pnlAmount = collateralAmount.plus(bigDecimalToBigInt(pnl));
+
+  const pairStorageContract = PairStorage.bind(
+    NetworkConfigs.getPairStorageAddress()
+  );
+  const closeFeePCall = pairStorageContract.try_pairCloseFeeP(pairIndex);
+  if (closeFeePCall.reverted) {
+    log.error("[closeFeePCall reverted] hash: {} pairIndex: {}", [
+      event.transaction.hash.toHexString(),
+      pairIndex.toString(),
+    ]);
+    return;
+  }
+  const closeFeeP = closeFeePCall.value;
+  const closingFee = bigDecimalToBigInt(
+    leveragedAmount
+      .times(closeFeeP)
+      .toBigDecimal()
+      .div(PRECISION.times(BIGINT_HUNDRED).toBigDecimal())
```

Comment:
Maybe `PRECISION.times(BIGINT_HUNDRED).toBigDecimal()` should just be stored as `PRECISION_BD` in cosntants.ts?